tasks/main.yml

```diff
@@ -23,7 +23,7 @@
     set_fact:
       create_vms: "{{ create_vms | default([]) + [item]}}"
     with_items: "{{ vms }}"
-    when: (item.state is defined and item.state != 'absent') or (item.profile.state is defined and item.profile.state != 'absent') or (item.state is undefined and item.profile.state is undefined)
+    when: (item.state is defined and item.state != 'absent') and (item.profile.state is defined and item.profile.state != 'absent') or (item.state is undefined and item.profile.state is undefined)
```

Comment:
So now in case `state=absent` is found in profile or vm defintion we never create them. I think we should create them in case for example state is absent for profile but present for the vm definition. Maybe user could remove all vms from the profile except one. We currentl logic this won't work, right?